solutions/cses-1163.mdx

```diff
@@ -61,47 +62,152 @@ int main(){
 }
 ```
 
-## Solution 2 - Reversing the steps
-
-I found another solution by reversing all the steps and calculating the maximum
-length at each step. Here is my implementation.
-
+</CPPSection>
+</LanguageSection>
+
+	
+## Solution 2 - Going Backwards
+	
+We'll start by trying to find the maximum gap once all the traffic lights are
+added. This is the last thing we'll output, so we'll add it to the end of our
+output array. Then, we'll remove traffic lights in the reverse order to how they
+were added, and find the gap each removal creates.
+
+This gap is just the distance between the two street coordinates (either a
+traffic light or the beginning or ending of the street) next to a given traffic
+light stored in our set, so we can use our set to find these values and subtract
+them to get our gap.
+
+But this gap may not be the maximum gap. We'll compare this to the gap we found
+once all the traffic lights are added, and set the max gap to the greater value.
+We'll then set the next lowest element in the output array to this value, which
+will represent the greatest gap prior to adding the traffic light we just
+removed.
+	
+## Implementation
+**Time Complexity:** $\mathcal{O}(n)$
+	
+<LanguageSection>
+<JavaSection>
+
+<Warning>
+We have to use some nonstandard io (input/output) to pass all test cases in the allowed time on CSES, because of its slow grader. 
+</Warning>
+```java
+import java.io.*;
+import java.util.*;
+
+public class TrafficLights {
+	public static void main(String[] args) throws IOException {
+		BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
+		PrintWriter pw = new PrintWriter(System.out);
+		StringTokenizer st = new StringTokenizer(r.readLine());
+
+		int streetLength = Integer.parseInt(st.nextToken());
+		int lightNum = Integer.parseInt(st.nextToken());
+
+		// Using an array to read values since we can't get values from sets in Java
+		int[] opArray = new int[lightNum];
+		NavigableSet<Integer> streetPositions = new TreeSet<>();
```

Comment:
```suggestion
		Set<Integer> streetPositions = new TreeSet<>();
```

Reply:
This won't work - only NavigableSet has .lower and .higher

Reply:
mb didnt see